build.sbt

```diff
@@ -42,7 +42,9 @@ inThisBuild(Def.settings(
   },
   scalafixScalaBinaryVersion := scalaBinaryVersion.value,
   apacheSonatypeProjectProfile := "pekko",
-  versionScheme := Some("semver-spec")))
+  versionScheme := Some("semver-spec"),
+  // TODO: Remove when Pekko has a proper release
+  updateOptions := updateOptions.value.withLatestSnapshots(false)))
```

Comment:
We'll have staged releases to test soon. This change would have been useful in the past. Now its benefit is reduced.

Reply:
We are currently effectively DDOSing the Apache snapshots repo so I think this takes precedence

Reply:
Can you remove the TODO? This change is probably fine as long as we keep using sbt-dynver

Reply:
* you occasionally might need to try out a pre-release change in one of our downstream repos
* and then you have to remember to add back `updateOptions := updateOptions.value.withLatestSnapshots(false))`

Reply:
Sure, honestly the real problem here is that `updateOptions.value.withLatestSnapshots(false))` is not granular. In an idea world you would be able to set `updateOptions.value.withLatestSnapshots(false))` for artifacts that are only download remotely where we can verify that snapshots are immutable since they are version with git hashes from `main` branch.

I think we are scraping the bottom of the barrel a bit here, this is meant to be a temporary work around so we don't overload the ASF repo with requests and while there may be arguments for leaving `updateOptions.value.withLatestSnapshots(false))` after release there is also a reason why the default of `withLatestSnapshots` is `true` in that its incredibly frustrating/surprising when you actually happen to have mutable snapshots. The overloading of requests is not going to be an issue if a person occasionally uses snapshots locally, its an issue if we have 12+ repo's building nightly snapshots resolving against pekko snapshots.

@pjfanning Is this PR still a real blocker for you? We can always revert/revisit this later but the equivalent of these changes have already been merged in other pekko modules.